# Домашнє завдання до теми "Основи роботи з рядками та математичні операції"

1. Створіть змінні `a`_ `b`, `c` та `d`, задайте їм значення відповідно 101, 210, 111, 402, та обчисліть з допомогою Python наступний вираз:  $$ \frac{(a+b) * c}{d} $$

**Важливо**: тут ми маємо обчислити значення саме з використанням змінних, а не просто чисел, як би ми робили у калькуряторі. Змінні потрібні аби за бажання ми могли змінити їх значення та не переписувати код обчислення виразу, бо він буде працювати для різних значень змінних.

In [1]:
a = 101
b = 210
c  = 111
d = 402

((a + b) * c) / d

85.8731343283582

Очікувана відповідь: 85.8731343283582

2. Якою буде остача від ділення числа 11283 на 89? Обчисліть з допомогою Python.

In [6]:
11283 % 89

69

3. Скільки цілих разів поміститься число 89 в 11283? (Або що те саме: який буде результат цілочисельного ділення 11283 на 89?)

In [7]:
11283 // 89

126

4. Запишіть у змінну `name` ваше імʼя.

In [9]:
name = "Mykola"

5. Запишіть у змінну `surname` ваше прізвище.

In [10]:
surname = "Kalashnyk"

6. Сконкатенуйте імʼя та прізвище в одну фразу через пробіл. Запишіть результат у нову змінну `name_surname` та виведіть результат на екран.  

In [13]:
print(name + " " + surname)

Mykola Kalashnyk


Очікувана відповідь (тільки з вашими даними): 'Hanna Pylieva'

7. Зробіть усі літери змінної `name_surname` заглавними використовуючи метод рядків в Python:

In [16]:
# First variant

name_surname = "Mykola Kalashnyk"
name_surname.upper()

# Second variant

name_surename = name + " " + surname
name_surename.upper()
print(name_surename.upper())

MYKOLA KALASHNYK


Очікувана відповідь (тільки з вашими даними): 'HANNA PYLIEVA'

8. Виведіть третю літеру вашого `name_surname`. Тут уважно, треба саме третя літера. Згадайте, з якого значення починається індексування значень в Python.

In [74]:

name_surname[2]

'k'

Очікувана відповідь (тільки з вашими даними): 'n'

9. Виведіть першу та останню літеру вашого прізвища.

In [48]:
display(len(name_surname))
print(name_surname[0])
print(name_surname[15])

9

Очікувана відповідь (тільки з вашими даними): ('P', 'a')

10. Виведіть довжину вашого прізвища.

In [73]:
len(surname)

4


Очікувана відповідь (тільки з вашими даними): 7


11. Виведіть кількість літер `a` у столиці Мадагаскару `Antananarivo`. Важливо врахувати всі літери, як великі, так і маленькі.

In [ ]:
capital = "Antananarivo"
capital_low = capital.lower()
capital_sum_a = capital_low.count("a")
print(capital_sum_a)

Очікувана відповідь: 4

12. Створіть рядок `my_string` зі значенням: "The quick brown fox jumps over the lazy dog!!!".   
Перевірте, чи починається рядок `my_string` з "The" і чи закінчується він "dog!". Виведіть результати на екран.

In [87]:
# First variant
my_string = "The quick brown fox jumps over the lazy dog!!!"
display(len(my_string))
print(my_string[0:3])
print(my_string[43:46])

# Actual result:
# The first 3 characters of the string : “The”
# Last 3 characters of the string : “!!!”

# 2nd var (with methods approach)
my_string = "The quick brown fox jumps over the lazy dog!!!"
print(my_string.startswith("The"))
print(my_string.endswith("dog!"))

46

The
!!!
True
False


13. Розділіть рядок `my_string` на список окремих слів, а потім з'єднайте всі слова через нижнє підкреслювання. Виведіть результат на екран.

In [86]:
my_string_split = my_string.split()
display(my_string_split)
s = "_"
print(s.join(my_string_split))

['The', 'quick', 'brown', 'fox', 'jumps', 'over', 'the', 'lazy', 'dog!!!']

The_quick_brown_fox_jumps_over_the_lazy_dog!!!


Очікуваний результат: `The_quick_brown_fox_jumps_over_the_lazy_dog!!!`